transports/janus_mqtt.c

```diff
@@ -0,0 +1,648 @@
+/*! \file   janus_mqtt.c
+ * \author Andrei Nesterov <ae.nesterov@gmail.com>
+ * \copyright GNU General Public License v3
+ * \brief  Janus MQTT transport plugin
+ * \details  This is an implementation of a MQTT transport for the Janus API,
+ * using the Eclipse Paho C Client library (https://eclipse.org/paho/clients/c).
+ * This means that this module adds support for MQTT based messaging as
+ * an alternative "transport" for API requests, responses and notifications.
+ * This is only useful when you're handling the communication with
+ * clients your own way. Right now, you can only configure
+ * the address of the MQTT broker to use, and the queues to
+ * make use of to receive (to-janus) and send (from-janus) messages
+ * from/to an external application. As with WebSockets, considering that
+ * requests wouldn't include a path to address some mandatory information,
+ * these requests addressed to Janus should include as part of their payload,
+ * when needed, additional pieces of information like \c session_id and
+ * \c handle_id. That is, where you'd send a Janus request related to a
+ * specific session to the \c /janus/<session> path, with RabbitMQ
+ * you'd have to send the same request with an additional \c session_id
+ * field in the JSON payload.
+ * \note When you create a session using RabbitMQ, a subscription to the
+ * events related to it is done automatically through the outgoing queue,
+ * so no need for an explicit request as the GET in the plain HTTP API.
+ *
+ * \ingroup transports
+ * \ref transports
+ */
+
+#include "transport.h"
+#include "../debug.h"
+#include "../config.h"
+#include "../utils.h"
+#include <MQTTAsync.h>
```

Comment:
Can you add an empty line here? We try to separate local includes from includes related to dependencies. Probably silly, but just coding style :smile: